### PR TITLE
[master < T0889-MG] Modify logaical operators to conform openCyper policies regarding checking against NULL in CASE expressions

### PR DESCRIPTION
**Pull request by gvolfing**
[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

### CODE REVIEW

**Comment by gvolfing** [comment]
Replacement PR for https://github.com/memgraph/memgraph/pull/422

File: tests/gql_behave/tests/memgraph_V1/features/case.feature [hunk 1]
```diff
@@ -57,3 +57,33 @@ Feature: Case
...
+
+    Scenario: Simple CASE nullcheck does not have match:
+        Given an empty graph
+        When executing query:
+            """
+            WITH 2 AS name RETURN CASE name WHEN 3 THEN 'works' WHEN null THEN "doesn't work" ELSE 'something went wrong' END
+            """
+        Then the result should be:
+            |  CASE name WHEN 3 THEN 'works' WHEN null THEN "doesn't work" ELSE 'something went wrong' END |
+            |  'something went wrong'                                                                      |
```
Comment (antaljanosbenjamin) [inline tests/gql_behave/tests/memgraph_V1/features/case.feature:69]:
```suggestion
            WITH 2 AS name RETURN CASE name WHEN 3 THEN 'something went wrong' WHEN null THEN "doesn't work" ELSE 'works' END
            """
        Then the result should be:
            |  CASE name WHEN 3 THEN 'something went wrong' WHEN null THEN "doesn't work" ELSE 'works' END |
            |  'works'                                                                      |
```

Seeing the expected output it is a bit confusing to see `something went wrong` as expected result, that's why I proposed these changes.

File: tests/gql_behave/tests/memgraph_V1/features/case.feature [hunk 2]
```diff
@@ -57,3 +57,33 @@ Feature: Case
...
+
+    Scenario: Generic CASE nullcheck does have match:
+        Given an empty graph
+        When executing query:
+            """
+            WITH 2 AS name RETURN CASE WHEN name is NULL THEN "doesn't work" WHEN name = 2 THEN "works" ELSE "something went wrong" END
+            """
+        Then the result should be:
+            |  CASE WHEN name is NULL THEN "doesn't work" WHEN name = 2 THEN "works" ELSE "something went wrong" END |
+            |  'works'                                                                                               |
```
Comment (antaljanosbenjamin) [inline tests/gql_behave/tests/memgraph_V1/features/case.feature:89]: I was thinking about adding a test case where the test expression itself is `null`. So `WITH null AS name RETURN CASE name WHEN null THEN "doesn't work" WHEN 2 "doesn't work" ELSE 'works' END`.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

**Review by antaljanosbenjamin — APPROVED** [review]

**Review by jbajic — APPROVED** [review]

**Comment by antaljanosbenjamin** [comment]
[Docs PR](https://github.com/memgraph/docs/pull/461)
